perf/change_point_detector.go

```diff
@@ -0,0 +1,79 @@
+package perf
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/evergreen-ci/cedar/util"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+type ChangeDetector interface {
+	DetectChanges([]float64) ([]ChangePoint, error)
+}
+
+type ChangePoint struct {
+	Index     int
+	Algorithm Algorithm
+}
+
+type Algorithm struct {
+	Name          string
+	Version       string
+	Configuration map[string]float64
```

Comment:
Having thought about this more, I'm pretty convinced that this ought to be a slice of key value pairs.

Reply:
Is that not just a map without the ability to look up by key in O(1) time? I may be missing something w.r.t. golang types.